dev/show_package_release_dates.py

```diff
@@ -0,0 +1,79 @@
+import subprocess
+import requests
+from concurrent.futures import ThreadPoolExecutor
+import traceback
+
+
+def get_distributions():
+    res = subprocess.run(["pip", "list"], stdout=subprocess.PIPE)
+    pip_list_stdout = res.stdout.decode("utf-8")
+    # `pip_list_stdout` looks like this:
+    # ``````````````````````````````````````````````````````````
+    # Package     Version             Location
+    # ----------- ------------------- --------------------------
+    # mlflow      1.21.1.dev0         /home/user/path/to/mlflow
+    # tensorflow  2.6.0
+    # ...
+    # ``````````````````````````````````````````````````````````
+    lines = pip_list_stdout.splitlines()[2:]  # `[2:]` removes the header
+    return [
+        # Extract package and version
+        line.split()[:2]
+        for line in lines
+    ]
+
+
+def get_release_date(package, version):
+    resp = requests.get(f"https://pypi.python.org/pypi/{package}/json")
```

Comment:
requests lib has a timeout default of 'None' which will indefinitely hold on to the connection until a server response arrives through the socket. Perhaps add a 10s timeout (or something?) to bomb this out if some packet loss occurs.

Reply:
Sounds good, thanks for pointing this out!